spec/grape/validations/validators/coerce_spec.rb

```diff
@@ -311,6 +311,22 @@ def self.parsed?(value)
           expect(last_response.status).to eq(400)
           expect(last_response.body).to eq('uri is invalid')
         end
+
+        context 'Array of types nil' do
+          it 'Empty array' do
+            subject.params do
+              requires :arry, type: Array[Integer]
+            end
+            subject.get '/array' do
+              params[:arry]
+            end
+
+            get '/array', arry: nil
```

Comment:
I don't see a default value anywhere here. You're passing a `nil`, and the API expects an array of integers. Feels like it should fail?

Reply:
I can add
```ruby
default: []
```
and it still fails on master. I found that since `arry` is an actual params, the default validator doesn't do anything. Actually it returns immediatly [here](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/validators/default.rb#L12)

Anyway, the default is kind of tricky for structures.
`Array`, `Set` and `Hash` will return a [default value](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/validators/coerce.rb#L76) even if not explicitly written in the param declaration

So these specs are working
```ruby
 [[Array, 'Array'], [Set, 'Set'], [Hash, 'ActiveSupport::HashWithIndifferentAccess']].each do |type, expected_class|
          context 'structures default value' do
            it 'Empty array' do
              subject.params do
                requires :arry, type: type
              end
              subject.get '/array' do
                params[:arry].class
              end

              get '/array', arry: nil
              expect(last_response.status).to eq(200)
              expect(last_response.body).to eq(expected_class)
            end
          end
        end
```
Therefore, I was expecting that an array of types with `nil` like `Array[Integer]` as param value would behave like a plain `Array`.

Reply:
Ok, please do add those specs and let's make sure to document this behavior?